### ISSUE TITLE
Doesn't work on mac os x

### ISSUE THREAD

**Issue by nashbridges** [d11wtq/oedipus#3]
Here's some OS related info:

Mac OS X 10.7.5
mysql: 5.6.10 (homebrew)
sphinxsearch: 2.0.6 (homebrew, built with PG support)
ruby: 1.9.3-p327
oedipus (0.0.16)

Index is built successfuly, and I can connect to daemon via mysql command line. But in irb I get errors:

```
s = Oedipus.connect '127.0.0.1:3312'
s[:index_name].fetch 3
NoMethodError: undefined method `each' for nil:NilClass
from /Users/nash/.rbenv/versions/1.9.3-p327/lib/ruby/gems/1.9.1/gems/oedipus-0.0.16/lib/oedipus/index.rb:250:in `block in meta_to_hash'
```

Sometimes it even hangs on `Oedipus.connect` step. I tried to launch deamon with `--console --logdebugvv`, but no output related to current issue was found.

Let me know, If you want more information.

**Comment by d11wtq** [comment]
@nashbridges that's interesting. I haven't tried building it against MySQL 5.6 yet (MySQL 5.5 works though), but perhaps the protocol implementation for multi-queries has changed. I'll try replicating your setup and get back to you. It appears that the issue is that Oedipus has executed two queries at the same time (your SphinxQL, plus SHOW META) and has only got one result back.

**Comment by nashbridges** [comment]
I've downgraded MySQL to 5.5.29, reinstalled Oedipus, got the same error. Also tried 5.5.29 with sphinx 2.0.5 (unfortunately, I couldn't compile versions earlier than that on Mac OS X) and got same results.

I have MySQL 5.5.29, sphinx 2.0.7 on my dev Ubuntu 12.04, and everything is working just fine.

If I set breakpoint in  https://github.com/d11wtq/oedipus/blob/master/lib/oedipus/index.rb#L236 :

```
pry(#<Oedipus::Index>)> stmts
=> ["SELECT * FROM some_index WHERE id = ?  ", "SHOW META"]
pry(#<Oedipus::Index>)> bind_values
=> [3]
pry(#<Oedipus::Index>)> records
=> [{"id"=>3,
  "weight"=>1,
  ...}]
pry(#<Oedipus::Index>)> meta
=> nil
```
